libretroshare/src/util/rsnet.cc

```diff
@@ -40,6 +43,72 @@
 	#define BIG_ENDIAN  4321
 #endif
 
+//https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-2
+#define DNSC_IN    1 //Internet (IN)
+//https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+#define DNST_A     1 //Ipv4 address
+#define DNST_AAAA 28 //Ipv6 address
+
+///Need to pack as we use sizeof them. (avoid padding)
+#pragma pack(1)
```

Comment:
Are you sure this work the same on every platform, like on embendedd ARM etc?

Reply:
It seems... But I can't test.

Reply:
> I think at least the implementation should go in it's own file as it is big enough,
No, as it use rsGetHostByName as fallback...